### DISCUSSION TITLE
How to customize the StableDiffusionXLPAGInpaintPipeline?

### DISCUSSION THREAD

**Discussion by LonglongaaaGo** [huggingface/diffusers#9462] [Q&A] [answered]
Hi, Can anyone can help with this?
I tried to customize the StableDiffusionXLPAGInpaintPipeline.
I first pip installed the newer diffuser version.
Based on the source code, I downloaded the  StableDiffusionXLPAGInpaintPipeline and tried to modify it by myself.
However, when I load pag packages like:
important diffusers.pipelines.pag.callbacks 

It shows it cannot find pag package.

How can I do this?
Thank you so much!

**Comment by LonglongaaaGo** [comment]
Hi @yiyixuxu ,
If you have time, I really appreciate that you can give some advice.
Thank you so much!

**Comment by asomoza** [comment]
Hi, the callbacks aren't inside the pag directory, when you customize a pipeline you have to make sure to match the dots with the correct subdirectories.

So in this case:

```python
from ...callbacks import MultiPipelineCallbacks, PipelineCallback
```

means that the callbacks are in the root of the project:

https://github.com/huggingface/diffusers/blob/5d476f57c58c3cf7f39e764236c93c267fe83ca1/src/diffusers/callbacks.py

which means you should import it like this if you have the pipeline outside of diffusers:

```python
from diffusers.callbacks import MultiPipelineCallbacks, PipelineCallback
```

**Reply by LonglongaaaGo** [reply]
Hi @asomoza 
Thank you so much for the help!
![image](https://github.com/user-attachments/assets/74f46326-d27a-4362-8dd3-937c8c2c244f)
When I directly import the package. It shows that it cannot find that.
Could you give me some advice?
Thanks!

**Reply by LonglongaaaGo** [reply]
My diffuser version is     0.31.0.dev0

**Reply by LonglongaaaGo** [reply]
![image](https://github.com/user-attachments/assets/88509aa7-19e1-4a52-b863-4ae5636958cb)

**Reply by asomoza** [reply, marked as answer]
both of those are correct and should work, if you can't import them there's something wrong with your installation or environment. The best way to resolve this is to create your environment again.

**Reply by LonglongaaaGo** [reply]
Hi @asomoza ,
Thanks!
 I try to install a new env. 
And let you know what happened

**Reply by LonglongaaaGo** [reply]
Wow, that works for me!
Awesome!
Thank you so much for the help! @asomoza
